device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -184,6 +185,33 @@ public void sendEventAsync(Message message, IotHubEventCallback callback, Object
         deviceIO.sendEventAsync(message, callback, callbackContext, this.config.getDeviceId());
     }
 
+    /**
+     * ASynchronously sends a batch of messages to the IoT Hub
+     * HTTPS messages will be sent in a single batch and MQTT and AMQP messages will be sent individually.
+     * Maximum payload size for HTTPS is 255KB
+     *
+     * @param messages the list of message to be sent.
+     * @param callback the callback to be invoked when a response is received.
+     * Can be {@code null}.
```

Comment:
Clarify that this callback is all-or-nothing for each batch message, so it will only be fired once